temporal-sdk/src/main/java/io/temporal/workflow/WorkflowInfo.java

```diff
@@ -136,6 +136,18 @@ public interface WorkflowInfo {
    */
   Optional<String> getParentRunId();
 
+  /**
+   * @return Workflow ID of the root Workflow
+   */
+  @Nullable
+  String getRootWorkflowId();
```

Comment:
This is a bit awkward since this can only be null on old servers that didn't have this value.

Reply:
Makes sense, may be worth documenting that here in the javadoc. I do appreciate that this is not using `Optional`.